R/control.R

```diff
@@ -72,9 +72,43 @@ control_grid <- function(
   res
 }
 
+# Helper function to print control settings using cli
+print_control_settings <- function(x, default = FALSE, defaults = NULL) {
+  # Get the fields to print
+  fields <- names(x)
+
+  # Optionally reduce to only non-defaults
+  if (default && !is.null(defaults)) {
+    fields <- fields[
+      !vapply(
+        fields,
+        function(field) {
+          identical(x[[field]], defaults[[field]])
+        },
+        logical(1)
+      )
+    ]
+  }
+
+  # Build formatted lines for each field
+  for (field in fields) {
+    value <- x[[field]]
+
+    if (is.function(value)) {
+      cli::cli_bullets(c(" " = "{.arg {field}}: <function>"))
+    } else if (inherits(value, "tune_backend_options")) {
+      cli::cli_bullets(c(" " = "{.arg {field}}: <backend_options>"))
+    } else {
```

Comment:
Can we add a branch for NULL values before the last that is: 

```r
 else if (is.null(value)) {
      cli::cli_bullets(c(" " = "{.arg {field}}: NULL"))
    }
```

We can then get the output without the empty spaces: 

```
Grid/resamples control object
  `verbose`: FALSE
  `allow_par`: TRUE
  `extract`: NULL
  `save_pred`: FALSE
  `pkgs`: NULL
  `save_workflow`: FALSE
  `event_level`: "first"
  `parallel_over`: NULL
  `backend_options`: NULL
  `workflow_size`: 100
```

Reply:
Done in latest commit